code/game/objects/items/reagent_containers/hypospray.dm

```diff
@@ -49,6 +49,8 @@
 		return
 	if(!istype(user))
 		return
+	if(!in_range(A, user))
```

Comment:
```suggestion
	if(!in_range(A, user) || !user.Adjacent(A))
```
First one checks if the item and the user are both on the same location relative to the attacked thing.
Second checks if the user is not just close to the thing, but if there are no barriers between them, such as a directional window or the likes.

Reply:
There is a `do_after` in this proc, which signals a timed action. Could you repeat this check after it, as the target may have moved while it finishes?
`if(!do_after(user, SKILL_TASK_EASY, TRUE, A, BUSY_ICON_UNSKILLED) || same checks)`

Reply:
done